hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactingMemStore.java

```diff
@@ -448,17 +455,28 @@ private void releaseResources() {
     * There is at most one thread per memstore instance.
     */
     private void doCompact() {
-
-      ImmutableSegment result = SegmentFactory.instance()  // create the scanner
-          .createImmutableSegment(getConfiguration(), getComparator(),
-              CompactingMemStore.DEEP_OVERHEAD_PER_PIPELINE_ITEM);
-
-      // the compaction processing
+      int cellsAfterComp = versionedList.getNumOfCells();
       try {
-        // Phase I: create the compacted MutableCellSetSegment
-        compactSegments(result);
-
-        // Phase II: swap the old compaction pipeline
+        // Phase I (optional): estimate the compaction expedience - CHECK COMPACTION
+        if (COMPACTION_PRE_CHECK) {
+          cellsAfterComp = countCellsForCompaction();
+
+          if (!isInterrupted.get() && (cellsAfterComp
+              > COMPACTION_TRIGGER_REMAIN_FACTOR * versionedList.getNumOfCells())) {
+            // too much cells "survive" the possible compaction we do not want to compact!
+            LOG.debug("Stopping the unworthy MemStore in-memory compaction for store "
```

Comment:
Makes sense to change the message. "Stopping to " --> "in-memory compaction does not pay off - storing the flattened segment".